include/Foundation/NSDecimal.h

```diff
@@ -1,62 +1,64 @@
-/* Copyright (c) 2007 Christopher J. W. Lloyd
-
-Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the
-"Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish,
-distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the
-following conditions:
-
-The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. */
-
-#ifndef _NSDECIMAL_H_
-#define _NSDECIMAL_H_
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+#pragma once
 
 #include <limits.h>
 #import <Foundation/NSObjCRuntime.h>
 
-#define NSDecimalMaxSize (8)
-#define NSDecimalNoScale SHRT_MAX
-
 enum { NSRoundPlain, NSRoundDown, NSRoundUp, NSRoundBankers };
-
-typedef uint32_t NSRoundingMode;
+typedef NSUInteger NSRoundingMode;
 
 enum { NSCalculationNoError = 0, NSCalculationLossOfPrecision, NSCalculationUnderflow, NSCalculationOverflow, NSCalculationDivideByZero };
+typedef NSUInteger NSCalculationError;
 
-typedef uint32_t NSCalculationError;
-
-typedef struct NSDecimal {
-    union {
-        signed int _exponent : 8;
-        unsigned int _length : 4;
-        unsigned int _isNegative : 1;
-        unsigned int _isCompact : 1;
-        unsigned int _reserved : 18;
-        unsigned short _mantissa[NSDecimalMaxSize];
-    };
+#define NSDecimalMaxSize (8)
+#define NSDecimalNoScale SHRT_MAX
 
-    double val;
+typedef struct {
```

Comment:
ah yes. updated.

---
In reply to: [80341079](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80341079) [](ancestors = 80341079)